### PR TITLE
10255 - Add premium account to non ministry feature UI changes

### PR DESCRIPTION
**Pull request by karthik-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/10255

*Description of changes:*

- Added child component access type that handles access type switch feature
- Refactor org service to add org patch endpoint
- test cases

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by sumesh-aot** [comment]
Check the code smells, we cannot have any new code smells introduced.

File: auth-web/src/components/auth/account-settings/account-info/AccountAccessType.vue [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+  $refs: {
+    accountAccessTypeForm: HTMLFormElement,
+    selectedAccessType: HTMLFormElement
+  }
+  private selectedAccessType: string = undefined
+  public AccessType = AccessType
+  public isLoading = false
+
+  public get currentOrgPaymentTypePad (): boolean {
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/account-settings/account-info/AccountAccessType.vue:121]: does isPad or some other meaningful method name?

File: auth-web/src/components/auth/account-settings/account-info/AccountAccessType.vue [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    return this.currentOrgPaymentType && this.currentOrgPaymentType === PaymentTypes.PAD
+  }
+
+  public get isCurrentOrgGovN (): boolean {
+    return this.organization.accessType === AccessType.GOVN
+  }
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/account-settings/account-info/AccountAccessType.vue:127]: move it to a common place

File: auth-web/src/components/auth/account-settings/account-info/AccountAccessType.vue [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  @Emit('update:viewOnlyMode')
+  cancelEdit () {
+    this.selectedAccessType = this.isCurrentOrgGovN ? AccessType.GOVN : AccessType.REGULAR
+    return {
+      component: 'accessType',
+      mode: true
+    }
+  }
+
+  public accessTypeMustBeGovN (): any {
```
Comment (saravanpa-aot) [inline auth-web/src/components/auth/account-settings/account-info/AccountAccessType.vue:151]: again method name

File: auth-web/src/services/org.services.ts [hunk 4]
```diff
@@ -88,8 +97,19 @@ export default class OrgService {
...
+      case PatchActions.UPDATE_STATUS:
+        return axios.patch(`${ConfigHelper.getAuthAPIUrl()}/orgs/${patchOrgPayload.orgIdentifier}`, { statusCode: patchOrgPayload.statusCode,
+          suspensionReasonCode: patchOrgPayload.suspensionReasonCode,
+          action: patchOrgPayload.action })
+      case PatchActions.UPDATE_ACCESS_TYPE:
+        return axios.patch(`${ConfigHelper.getAuthAPIUrl()}/orgs/${patchOrgPayload.orgIdentifier}`, { accessType: patchOrgPayload.accessType,
+          action: patchOrgPayload.action })
+      default:
+        break
+    }
```
Comment (saravanpa-aot) [inline auth-web/src/services/org.services.ts:111]: since axios.patch is common ,move it outside of the method and just make the request body diffenrent?

**Review by saravanpa-aot — APPROVED** [review]

**Pull request merged**